### PR TITLE
add a unit test job for cluster-baremetal-operator

### PR DESCRIPTION
**Pull request by dhellmann**
Depends on https://github.com/openshift/cluster-baremetal-operator/pull/5

### CODE REVIEW

**Comment by dhellmann** [comment]
/hold

This is going to need to be rebased after #11103 merges

**Review by sadasu — APPROVED** [review]
/lgtm

**Pull request merged**
